Octokit/Clients/IIssuesClient.cs

```diff
@@ -163,5 +163,25 @@ public interface IIssuesClient
         /// </param>
         /// <returns></returns>
         Task<Issue> Update(string owner, string name, int number, IssueUpdate issueUpdate);
+
+        /// <summary>
+        /// Locks an issue for the specified repository. Issue owners and users with push access can lock an issue.
+        /// </summary>
+        /// <remarks>https://developer.github.com/v3/issues/#lock-an-issue</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="number">The issue number</param>
+        /// <returns></returns>
+        Task LockIssue(string owner, string name, int number);
```

Comment:
Given this usage will look like this:

```
await github.Issue.LockIssue("owner", "name", 1);
```

Perhaps we can drop the `Issue` suffix here, to give:

```
github.Issue.Lock("owner", "name", 1);
```

Thoughts @haacked @ryangribble?